src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CollectionCodecs.java

```diff
@@ -103,14 +132,38 @@ static Set<Object> toCQLSet(
         continue;
       }
       if (elementCodec == null || !elementCodec.handlesJavaValue(element)) {
-        elementCodec = findElementCodec(valueCodecs, elementType, element);
+        elementCodec = findCollectionElementCodec(valueCodecs, elementType, element);
       }
       result.add(elementCodec.toCQL(element));
     }
     return result;
   }
 
-  private static JSONCodec<Object, Object> findElementCodec(
+  static Map<String, Object> toCQLMap(
+      List<JSONCodec<?, ?>> valueCodecs, DataType elementType, Map<?, ?> rawMapValue)
+      throws ToCQLCodecException {
+    Map<String, JsonLiteral<?>> mapValue = (Map<String, JsonLiteral<?>>) rawMapValue;
```

Comment:
Currently only support String key type, but eventually it support for other key data types by having different json structure. Should this code try to resolve key type codec and resolve key?

Reply:
I think that since we probably need re-designed codec for that case, adding key codec handling that we don't yet need is not probably worth doing, since it would probably need to be changed anyway. So I think I will defer that work until we add support for non-TEXT/ASCII keys.

Reply:
agree not needed yet, but this is a kind of ugly looking unchecked cast based on that assumption. can we add a comment that we are assuming only string keys because that is all we support now

Reply:
Yes, I will add a comment.